src/builder/builder.go

```diff
@@ -190,6 +231,29 @@ func (b *Builder) SignManifestMOM() {
 
 // UpdateRepo will fetch the clr-bundles for our configured Clear Linux version
 func (b *Builder) UpdateRepo(ver string, allbundles bool) {
+	pwd, err := os.Getwd()
+	if err != nil {
+		helpers.PrintError(err)
+		os.Exit(1)
+	}
+
+	// Make the folder to store rpms
+	err = os.Mkdir(pwd+"/rpms", 0755)
+	if os.IsExist(err) {
+		fmt.Println("Directory '/rpms' already exists. Skipping...")
+	} else if err != nil {
+		helpers.PrintError(err)
+		os.Exit(1)
+	}
+
+	err = os.Mkdir(pwd+"/local", 0755)
+	if os.IsExist(err) {
+		fmt.Println("Directory '/local' already exists. Skipping...")
+	} else if err != nil {
+		helpers.PrintError(err)
+		os.Exit(1)
+	}
+
```

Comment:
I don't think this logic can live here yet. It makes sense to me for these things to be be together eventually, but `UpdateRepo` needs to be refactored first.

Right now `UpdateRepo` does two things: 1) it downloads and unpacks a local cache of CLR bundle definitions, and 2) it populates/initializes the `mix-bundles` directory if it doesn't already exist.

Right now it's called in three places: 1) by `mixer init-mix`, 2) by `mixer get-bundles`, and 3) by `mixer add-bundles`

`UpdateRepo` is called in `AddBundles` to take advantage of the CLR bundle download/unpacking. `AddBundles` doesn't want the `mix-bundles` initialization, and takes advantage of the fact it'll skip it if it exists. This coupling is a bit dangerous. With this patch, it would attempt to add these  `rpm` and `local` directories every time you call `mixer add-bundles`, too.

Likewise, these directories would be re-added (and warn with a "already exists" message) anytime someone called `mixer get-bundles`, too. However, I am pushing for this command to be made obsolete, with the tool handling bundle caching automatically for you the way `AddBundles` does.

So really what we probably want is one higher-level (`helper.go`?) function that retrieves and unpacks bundles that can be called by both `UpdateRepo` and `AddBundles`, and then `UpdateRepo` could be thought of more as a "initialize all the directories" function.

Reply:
After reading your proposals for the tool I figured this was a bit out of place. Still, I don't want this PR to keep hanging until the refactor happens. So I think we have two options now: Leave it here and once there is a proper isolated function for creating the directories we move it along or isolate it in a separate function and once the refactor happens remember to move it to the appropriate location. I'm ok with either solution.

Reply:
I vote for the latter -- wrap the repo directory creation in a separate function and call it during init. 

And, actually, I think it needs to be an optional thing (created and defined in the conf only if you pass "--local-repo" or something like that). This is a little trickier than I realized. Yesterday @mdhorn reminded me that the code in `BuildChroots` checks if `b.Repodir !- ""` to determine how to use `m4` for expanding the yum config. (The section telling it about the local RPMS is only included if the ifdef passes.) This means that as it is right now, your code will *always* define these repos, and so `b.Repodir` will *never* be "", so `m4` will *always* include the local RPM section, even if it's empty.

@tmarcu, can you double check what I just said above?

Reply:
Yes, that is what happens @kevincwells . I didn't comment previously since it technically works still...it'll just make yum look into your local repo every time before checking upstream, but that's probably misleading a bit. I like the idea of a --local-repo flag, and/or just removing the code that creates /local and /rpms from this patch for now. The current documentation describes those 2 folders needing to be made, so it's not a big deal if we just keep that functionality as is, and then we can merge this patch and iterate on it afterwards.

Reply:
@tmarcu if you spot stuff like this, please definitely speak up, as they create dead paths in the code.

I'm not saying that these directories can't get created automatically (I like the idea of not having to make them) or that it's even a bad thing for YUM to look into your local repo every time (is it that much of an optimization to make it not?). But if we go that route, we definitely need to take out the check for these repos, the use of `m4`, and the dead paths in the code. One or the other.